src/references/script_debugger.md

```diff
@@ -0,0 +1,100 @@
+
+## Stepping Through a Script
+
+Stepping through a script is an important part of debugging, it allows creators to walk through each line of the script to inspect the variables and objects to see what is contained in them.
+
+### Step Into
+
+**Step Into** ![Step Into](../img/ScriptDebugger/step_into_icon.png)will be frequently used to step through a script line by line. This also includes functions. If a function is about to be called, and the code needs to be debugged in that function, then the next step is to go into that function and debug it line by line by using **Step Into**.
+{: .image-inline-text .image-background }
+
+### Step Over
+
+**Step Over** ![Step Over](../img/ScriptDebugger/step_over_icon.png)will execute as one complete step. For example, if the current line is a function, then stepping over that function will move to the next line instead of stepping through each line of that function.
+{: .image-inline-text .image-background }
+
+### Step Out
+
+**Step Out** ![Step Out](../img/ScriptDebugger/step_out_icon.png)is used to step out of a function. For example, when only part of a function needs to be debugged, stepping out will tell the **Script Debugger** to run the rest of that function.
+{: .image-inline-text .image-background }
+
+## Select a Script Task
+
+When a project gets bigger, more scripts are created to handle different parts of the game. Breakpoints can be added to different scripts, and the **Script Debugger** will pause execution for whichever script is running first that has a breakpoint set. Scripts that have not been completed can be stepped through. This can be done from **Script Task** drop down in the **Script Debugger** window.
+
+The **Script Task** drop down will list all the scripts, and also indicate which one is currently running, scheduled, and which have completed.
+
+![!Task List](../img/ScriptDebugger/task_list.png){: .center loading="lazy" }
+
+!!! Warning "Script Execution Order"
+    Do not rely on the script execution order in preview to be the same for the live version of your game. In some cases, scripts may not be in the order as you expect, especially with networked contexts.
+
+## Stack Frames
+
+The **Stack Frames** panel in the **Script Debugger** window, shows the function calls that are currently on the stack for the current running script task. The **Stack Frames** panel shows the order in which the functions are getting called, which is a good way to see the execution flow of the script for the current task.
+
+For example, in the picture below, you can see the execution flow for the current running script.
+
+![!Stack Frames](../img/ScriptDebugger/stack_frames.png){: .center loading="lazy" }
+
+## Learn More
+
+[Intro to Scripting](../tutorials/scripting_intro.md) | [Lua Scripting Tutorial](../tutorials/lua_basics_helloworld.md) | [Lua Scripting Tutorial, Part 2](../tutorials/lua_basics_lightbulb.md) | [Advanced Scripting in Core](../tutorials/race_timer.md)
```

Comment:
Cut Lua Script Tutorial 2, and add links to the Performance Profiler (another thing you would use to diagnose problems) and AI Activity System (has its own debugger-like interface).